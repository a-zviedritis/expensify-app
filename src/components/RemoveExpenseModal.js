import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = (props) => (
  <Modal 
    isOpen={!!props.expenseForRemoval}
    onRequestClose={props.handleDenyRemoval}
    contentLabel="Remove expense"
    closeTimeoutMS={50}
    className="modal"
  >
    <h3 className="modal__title">Remove expense?</h3>
    <div className="modal__selection">
      <button className="button" onClick={props.handleConfirmRemoval}>Yes</button>
      <button className="button" onClick={props.handleDenyRemoval}>No</button>
    </div>
  </Modal>
);

export default RemoveExpenseModal;