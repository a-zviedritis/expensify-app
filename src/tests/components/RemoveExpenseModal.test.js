import React from 'react';
import { shallow } from 'enzyme';
import RemoveExpenseModal from '../../components/RemoveExpenseModal';
import expenses from '../fixtures/expenses';

let handleDenyRemoval, handleConfirmRemoval, history, wrapper;

beforeEach(() => {
  handleDenyRemoval = jest.fn();
  handleConfirmRemoval = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <RemoveExpenseModal 
      handleDenyRemoval={handleDenyRemoval} 
      handleConfirmRemoval={handleConfirmRemoval} 
      history={history} 
      expenseForRemoval={expenses[0]}/>
  );
});

test('should render RemoveExpenseModal correctly', () => {
  expect(wrapper).toMatchSnapshot();
});