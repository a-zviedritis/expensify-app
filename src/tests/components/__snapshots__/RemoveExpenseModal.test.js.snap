// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveExpenseModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={50}
  contentLabel="Remove expense"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h3
    className="modal__title"
  >
    Remove expense?
  </h3>
  <div
    className="modal__selection"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</Modal>
`;
